packages/frontend/src/hooks/useCartesianChartConfig.ts

```diff
@@ -523,13 +517,22 @@ const useCartesianChartConfig = ({
         [dirtyLayout, dirtyEchartsConfig],
     );
 
-    const chartType =
-        dirtyChartType === CartesianSeriesType.LINE && areaStyle
-            ? CartesianSeriesType.AREA
-            : dirtyChartType;
+    const { dirtyChartType } = useMemo(() => {
```

Comment:
`dirtyChartType` is now always calculated based on the first series type and we don't need to manage its state. The same for the `areaStyle`, which was completely removed.